sindresorhus/refined-github
# Add `sync-pr-commit-title` feature (replaces `fix-squash-and-merge-title`)

Fixes https://github.com/sindresorhus/refined-github/issues/1790
Closes https://github.com/sindresorhus/refined-github/issues/1743

- When the commit title is manually updated, RGH will offer to also update the PR title
- Title/message features merged into one
- The title box is focused after it's opened
- The message box now expands to fit the content
- This now applies to Merge commits too

# Next

- [x] Add migration in background.js
- [x] Should we drop “fix-squash-and-merge-message”? I'm not entirely sure the new message is useful anyway and it contains a lot of noise ("fixes this and that", "I tried this", "What do you think", PR template comment, etc)
- [x] Does the "PR title update" part belong to the same feature? There's some overlap and both features kinda do the same thing, in two directions

# Test 

<details>
<summary>Commit info update</summary>


1. Visit PR with ONE commit (with 2+ commits, GitHub already updates **the title**)
2. Open the merge box (as Squash or Merge)
---
1. Visit PR
2. Change PR title
2. Open the merge box (as Squash or Merge)


</details>

<details>
<summary>PR title update</summary>


1. Visit PR
2. Open the merge box (as Squash or Merge)
3. Change title (the message will appear)
4. Undo changes (the message will disappear)
5. Merge (the PR title will be updated if the message was shown)
---
1. Visit PR
2. Open the merge box (as Squash or Merge)
3. Change title (the message will appear)
4. Press `cancel` (the message will disappear)
5. Merge (the PR title will **not** be updated)

</details>

**Untested**

Where admin permissions are required
